metricbeat/module/beat/stats/data.go

```diff
@@ -76,8 +76,29 @@ var (
 			"pipeline": c.Dict("pipeline", s.Schema{
 				"clients": c.Int("clients"),
 				"queue": c.Dict("queue", s.Schema{
-					"acked":      c.Int("acked"),
 					"max_events": c.Int("max_events"),
+
+					"added": c.Dict("added", s.Schema{
+						"events": c.Int("events"),
+						"bytes":  c.Int("bytes"),
+					}),
+					"consumed": c.Dict("consumed", s.Schema{
+						"events": c.Int("events"),
+						"bytes":  c.Int("bytes"),
+					}),
+					"removed": c.Dict("removed", s.Schema{
+						"events": c.Int("events"),
+						"bytes":  c.Int("bytes"),
+					}),
+					"filled": c.Dict("filled", s.Schema{
+						"events": c.Int("events"),
+						"bytes":  c.Int("bytes"),
+						"pct":    c.Float("pct"),
+					}),
+
+					// Backwards compatibility: "acked" is the old name for
+					// "removed.events" and should not be used by new code/dashboards.
+					"acked": c.Int("acked"),
```

Comment:
Maybe I am misreading the connection between the fields, but it seems like this value is not going to ever return anything given that the `queue.acked` value was removed above, so keeping it in the schema seems wasteful?

I think that there is a feature of the schema that allows you to bring another value into this field, but I have not had a need for it, so I am not certain.

Reply:
Oh, yes sorry that's a misunderstanding: the removed things above are all dead code. The real metrics are handled in `libbeat/publisher/queue/monitoring.go` and include `max_events` and `acked` already. But the dead fields sent me on a wild goose chase during this fix, so I am removing them to avoid future confusion :-)